internal/metrics/metrics_test.go

```diff
@@ -0,0 +1,5 @@
+package metrics
+
+import "github.com/lunarway/strong-duckling/internal/tcpchecker"
+
+var _ tcpchecker.Reporter = &PrometheusReporter{}
```

Comment:
Could we add some tests for the logic in `ReportPortCheck`? Multiple nested `if` statements are hard to maintain without along with verifying edge cases.